xla/service/gpu/cudnn_fused_mha_rewriter.cc

```diff
@@ -49,6 +52,11 @@ auto OptionalConvert(Pattern pattern) {
   return m::AnyOf<HloInstruction>(m::Convert(shared), shared);
 }
 
+template <typename Pattern>
+auto OptionalBitcast(Pattern pattern) {
```

Comment:
Those tend to be super-expensive to compile, cf. the fix in other mha rewrite PR (for inference)

Reply:
i accidentally added std::move to all subpatterns, reverted my changes

Reply:
Is the compile-time concern resolved?

Reply:
yes, this is kept the same as the other sharedPattern templates